### ISSUE TITLE
NVDA unable to track cursor in the GMail message composer when using Chrome

### ISSUE THREAD

**Issue by ssawczyn** [nvaccess/nvda#6424]
When composing a new Email or replying to an Email within GMail using Chrome, NVDA does not properly track the cursor.  Specifically, NVDA does not read the character on which the cursor is located.  To replicate this:
1. Using Chrome open GMail in standard view.
2. Compose a new message.
3. Type some text in the message area.
4. Attempt to edit text.

**Comment by jcsteh** [comment]
Can you be more specific as to exactly what you're seeing here? When you
"attempt to edit text", what exactly are you pressing and what does NVDA
report? if you use the left and right arrow keys, does NVDA report nothing
or does it report the incorrect character (e.g. the one before/after)?

Cursor tracking does seem to be a bit unreliable for me in Chrome stable.
It is a lot better in Chrome Canary, though. However, even in Chrome
stable, I still get characters reported, though often the character I was
previously on instead of the character at the new position.

**Comment by feerrenrut** [comment]
Assuming we understand the issue, and Jamie has described it correctly. We think this issue is improving with updates to Chrome, while annoying this seems to be reasonably usable in Chrome canary. This would be a large chunk of uncertain work to work around in nvda. Applying a P3 label. Any questions or clarifications welcome.

**Comment by feerrenrut** [comment]
Technical:
A quick approach to reducing the frequency that this occurs could be to increase the time out used to wait for the caret to move.

**Comment by iubito** [comment]
Hi!
The problem is still present, using Chrome-based or Firefox browser.
It appear that sometimes, a ZWSP (Zero-width space, U+200B,decimal & # 8203;) is inserted, and then NVDA cant read the line while moving caret or mouse pointer.
Selecting a word or a line can say nothing or the selected text.

If you are lucky, in some rare case, moving the caret to a line containing a ZWSP will say "?" (interrogation mark).

Hope this help to find a bugfix for ZWSP in browser textarea, at least in Gmail.
ZWSP doesn't disturb NVDA in Notepad++.

**Comment by seanbudd** [comment]
@iubito - please open a new issue and fill out the bug template.
